packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -441,6 +442,26 @@ export class DataStores implements IDisposable {
         return builder.getSummaryTree();
     }
 
+    public async getGCData(): Promise<IGCData> {
+        const builder = new GCDataBuilder();
+        // Iterate over each store and get its GC data.
+        await Promise.all(Array.from(this.contexts)
+            .filter(([_, context]) => {
+                // GC can only run on clients with no local changes!
+                assert(context.attachState !== AttachState.Attaching, "Not expecting local data stores during GC");
```

Comment:
I'm not 100% sure this assert is correct. Did you want to do != AttachState.Detached ?
Also should it be a check, not assert? I.e. it's totally normal to have unattached data stores

Reply:
I borrowed this logic from summarize. I reckon its because summarize / GC should only run in summarizer client which cannot have local contexts / channels.
Now that I think about it, it does not seem correct because summarize is a public API and it can be called on clients with local state too. Also, snapshot API in Container eventually calls this which might be problem if called on a Container that has local state.

Reply:
But what happens if a data store / channel context is detached or attaching? Do we just ignore it for snapshot purposes? That also does not seem right.

Reply:
For GC, not including detached stores / channels makes sense because they are not connected to the graph. So any references it might have are not connected to rest of the system.